### DISCUSSION TITLE
The first time you run the programme, you get an error!!!  Help!

### DISCUSSION THREAD

**Discussion by cccp421** [Mayuresh1611/Paper-Piano#8] [Q&A] [answered]
Hello building owner. I have a problem with `model.predict(data)` in the `Predict` function in `model.py` when I run your file, how can I change it?  
It's when I run the `run.py` file for the first time, because I don't have the training weights file `xxx.h5`, and when I execute the guide `RUN` function, it causes an error in the programme because it calls the `model.py` file.

**Comment by Mayuresh1611** [comment]
Hi! sorry to responding late. Can you share the error message? 
Also you can try one thing, delete the ```touch_detection_model.h5``` file from ```models``` folder. Then try rerunning the ```run.py```.
let me know if it fixes or not

**Reply by cccp421** [reply]
The error occurs when running the `run.py` file without the `touch_detection_model.h5` file. The specific location of the error is in the `model.py` file.

**Reply by cccp421** [reply]
When the 'touch_detection_model.h5 'file exists, running the 'run.py' file is normal and does not produce any errors.

**Reply by cccp421** [reply]
<img width="1178" alt="04a39d35b907fa3ddc82a3d9e1b3fce" src="https://github.com/Mayuresh1611/Paper-Piano/assets/149547708/28b16b88-e378-46af-8ccb-5c017460c91e">
This error is generated exactly by re-running the `run.py` file after deleting the `touch_detection_model.h5` file

**Reply by Mayuresh1611** [reply]
Its worse to say, but it runs fine on my side. Is there any problem with the pycharm itself? I don't get the exact cause of this error, it is the warning? but it doesn't have anything to do with imo.

**Reply by cccp421** [reply, marked as answer]
Hello! I seem to know the reason for the error in the programme. I tried to make changes in the program that called the `model.py` file, there are two places: one is the `RUN` function in the `run.py` file, and the other is the function `predict_worker` that calls the multithreading in the `piano.py` file as well as the function `start_piano`. i first commented out the `RUN` function and don't execute the `RUN` function.
Then I found that when the file `touch_detection_model.h5` does not exist in the models folder, executing the `run.py` program will train first, get the file `touch_detection_model.h5`, and then call multithreading. Since `model.py` just started to first determine that the `touch_detection_model.h5` file does not exist, resulting in the `model.predict(data)` in the Predict function does not exist, thus triggering the error report.
In my opinion, this error occurs due to a problem with the placement of the if judgement statement in the `model.py` file

**Reply by Mayuresh1611** [reply]
I believe it too be true to, how about you create a branch and make pull request.

**Reply by cccp421** [reply]
Thanks! I'm currently attempting to deploy your project on a `Jetson Nano` development board, with a lot of program modifications, but there are still quite a few issues that need to be resolved. If it's possible, when I finish the deployment, I will make it public in my Github repository and indicate the source of the programme.

**Reply by Mayuresh1611** [reply]
Wow! That is quite ambitious, I am eager to have look at it. I am also working on better version, I will let you know when I start hitting the keyboard. Currently just theoriotising how effectively can it be made.

**Reply by cccp421** [reply]
Haha, I'd like to see a better version of `Paper-Piano` too.
